### PR TITLE
[2/6] [oxql] move core types to a new oxql-types crate

### PR DESCRIPTION
**Pull request by sunshowers**
Means that the Nexus external API crate doesn't have to pull in oximeter-db.

I did have to expose a few mutators, but my general reasoning is that because these types derive `Deserialize`, it's always possible to make these changes by serializing into JSON and working with them there.

Depends on #6355.

### CODE REVIEW

File: openapi/nexus.json [hunk 1]
```diff
@@ -20131,10 +20131,20 @@
         "type": "object",
         "properties": {
           "metric_type": {
-            "$ref": "#/components/schemas/MetricType"
+            "description": "The type of this metric.",
```
Comment (bnaecker) [inline openapi/nexus.json:20134]: Interesting, I wonder why this changed?

File: oximeter/oxql-types/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+use chrono::{DateTime, Utc};
+use std::time::Duration;
+
+pub mod point;
+pub mod table;
+
+pub use self::table::Table;
+pub use self::table::Timeseries;
+
+pub type TimeseriesKey = u64;
```
Comment (bnaecker) [inline oximeter/oxql-types/src/lib.rs:16]: This type is really about the data storage in ClickHouse. Ideally it'd be defined in `oximeter_db`, but I get why you need to move it down a level in the dependency tree. Maybe we could put it alongside the other schema-related types, like `TimeseriesSchema`, which I think as of #6355 is part of `oximeter_types::schema`?

**Review by bnaecker — COMMENTED** [review]
Wow, this one looks non-trivial, thanks for digging into it! I have one request, that we move `TimeseriesKey` around a bit, since it's more than an OxQL concept. Other than that, LGTM!

File: oximeter/oxql-types/src/lib.rs (hunk 2 above)
Comment (sunshowers) [reply oximeter/oxql-types/src/lib.rs:16]: Sure, that sounds good to me.

File: oximeter/oxql-types/src/point.rs [hunk 3]
```diff
@@ -169,10 +169,10 @@ impl CumulativeDatum {
...
 #[derive(Clone, Debug, Deserialize, JsonSchema, PartialEq, Serialize)]
 pub struct Values {
-    // The data values.
-    pub(super) values: ValueArray,
-    // The type of this metric.
-    pub(super) metric_type: MetricType,
+    /// The data values.
+    pub values: ValueArray,
+    /// The type of this metric.
+    pub metric_type: MetricType,
```
Comment (sunshowers) [inline oximeter/oxql-types/src/point.rs:175]: > Interesting, I wonder why this changed?

Ah I added doc comments here which caused that change to happen. There are no semantic changes to the OpenAPI document.

File: openapi/nexus.json (hunk 1 above)
Comment (sunshowers) [reply openapi/nexus.json:20134]: See https://github.com/oxidecomputer/omicron/pull/6364#discussion_r1720579988.

File: oximeter/oxql-types/src/lib.rs (hunk 2 above)
Comment (sunshowers) [reply oximeter/oxql-types/src/lib.rs]: Done.

**Review by bnaecker — APPROVED** [review]
Looks good, thanks for moving over the `TimeseriesKey` type!

**Pull request merged**